syntax = "proto3";
package solana_corecast;

import "corecast/request.proto";
import "corecast/stream_message.proto";


service CoreCast {
  rpc DexTrades (SubscribeTradesRequest) returns (stream DexTradeStreamMessage);
  rpc DexOrders (SubscribeOrdersRequest) returns (stream DexOrderStreamMessage);
  rpc DexPools  (SubscribePoolsRequest)  returns (stream PoolLiquidityChangeStreamMessage);

  rpc Transactions (SubscribeTransactionsRequest) returns (stream ParsedTransactionStreamMessage);
  rpc Transfers    (SubscribeTransfersRequest)    returns (stream TransferStreamMessage);
  rpc Balances     (SubscribeBalanceUpdateRequest) returns (stream BalanceUpdateStreamMessage);
}